### DISCUSSION TITLE
Units for `MMEAN_NPPDAILY_CO`, `MMEAN_TRANSP_CO`, and other cohort-level outputs

### DISCUSSION THREAD

**Discussion by Aariq** [EDmodel/ED2#381] [Q&A] [answered]
I'm just trying to confirm what the units are for some variables found in the -E- file outputs.  According to the metadata, `MMEAN_NPPDAILY_CO` is in kgC/m2/yr and `MMEAN_TRANSP_CO` is in kg/m2/s.  Assuming these units are correct has led me to some values that seem to be off by orders of magnitude.  They also aren't internally consistent with the polygon-level variables in the same -E- file.  For example, in a file with one site, one patch, and two cohorts the cohorts have `MMEAN_NPPDAILY_CO` values of 1.56 and 2.98.  In the same -E- file, `MMEAN_NPPDAILY_PY` is 0.906.  I would think that one would add the cohort-level values to get the polygon level values, but that's obviously not how it works.  The -T- file generated for the same run has hourly values for NPP in `FMEAN_NPP_PY` and if I take a monthly average it's 0.478, which doesn't match the monthly mean in the -E- file.  What's going on here?

**Comment by Aariq** [comment, marked as answer]
Found one piece of the puzzle here: https://github.com/EDmodel/ED2/issues/342

So the units for `MMEAN_NPPDAILY_CO` are actually kgC/plant/yr.  Multiplying by `NPLANT` (plant/m^2) gives the correct units, but the numbers still don't add up.  I get 0.156 and 0.296 for the two cohorts which sums to 0.451.  That's pretty close to what's in the -T- file, but still not very close to `MMEAN_NPPDAILY_PY` (0.906)

### UPDATE:
Ok, I tried the math with a different -E- file and got it to work

$$
NPP_{polygon} = \sum NPP_{i,k} * NPLANT_{i,k} * AREA_{k}
$$

where i = cohort and k = patch.

**Reply by crollinson** [reply]
I haven't worked with cohort-scale NPP in a while, but in general in my experience, _CO vars are in per plant and you need to do the scaling by density that you describe above.  To bring that to the PY scale, you'll have to also then weight by the patch area.  

FWIW, trying to bring cohort-scale data to the site/polygon scale is always a giant pain.  For some variables there are things labeled with PFT (and PY?) and those are often broken down by PFT and size class if that's what you're going for.

I'm a bit fuzzy on some of this by now, but hopefully this helps point in the right direction.

**Reply by Aariq** [reply]
I really just want variables broken down by PFT.  The only way I could figure out how to do that was to start from the cohort-level variables ending in `_CO` and sum by PFT.  I'm just using the `_PY` variables and the -T- files to check my math.  I was hoping things would add up, but they don't.  In the example above there is only one patch so it's not a problem of weighting by patch area, but thanks for pointing that out.

**Reply by ashehad** [reply]
I think you need to multiply the sum of 0.156 and 0.296 by ncohort value. It will be hopefully 0.906. 
Note: ncohort is PACO_N, I think. In your case it is 2, right?

**Reply by Aariq** [reply]
Ok, I tried the math with a different -E- file and got it to work (without multiplying by PACO_N because that's just the number of values in the patch)

$$
NPP_{polygon} = \sum NPP_{i,k} * NPLANT_{i,k} * AREA_{k}
$$

where i = cohort and k = patch.
